.github/workflows/ci.yml

```diff
@@ -14,45 +14,38 @@ jobs:
     strategy:
       matrix:
         toolchain: [stable, nightly]
-        os: [windows-2019, ubuntu-20.04, macos-10.15]
+        os: [windows-latest, ubuntu-latest, macos-latest]
```

Comment:
I feel these lines should be reverted. Opting unconditionally into the latest runtime environment available is demonstrably unwise for reproducible builds and a stable pipeline over time. It will generate surprise failing builds over time, including on unchanged branches from day to day. It's not a given that rapid support for the latest OS is a project goal, and there are other ways to march towards that less disruptively if it is. 

This also wasn't in scope for the comment by Cart that was cited.

Reply:
> This also wasn't in scope for the comment by Cart that was cited.

Correct, this was an extra change that I felt would be beneficial, like most of the changes in this PR.

In my experience, pinning CI the latest stable versions of things has been the lowest friction option for in-development projects.  Still, it's six of one, half a dozen of the other. I can certainly revert this.

It would be helpful to know Bevy's stance on supported versions of things.  For example, we can simplify the `clippy` command in this file if we're willing to _not_ support versions of Rust older than 1.45. In general, the attitude I've observed is "we're in a rapidly evolving state, we'll slow down and maintain support for things after 1.0" -- which I (perhaps incorrectly) interpreted as supporting primarily the newest versions of things, including languages and OS's.  @cart what are your actual feelings on the matter?

Reply:
As a counterpoint, it's broadly considered to be an antipattern to use `latest` tags in publishing or consuming Docker images, and that's spiritually very similar. You don't typically see `FROM ubuntu:latest` in practice there, you see it pinning to a major release as the original content here did. But agreed, Cart's opinion is the only one that matters in the end.

Reply:
Yeah this is a tough call. As someone who has run "docker services" professionally in the past, I can attest to not wanting to use "latest" in that context. Its just too easy for things to accidentally break in an uncontrolled way. You can't afford to take that risk in the context of a high-traffic service.

_However_ in a professional context, I also had a compliance team (and automation) that would yell at us when we had any servers running that were "too old". We don't have that here. And I have approximately zero motivation to build that automation or even to periodically manually check for updates. (nor do I want to get pinged every time ubuntu, windows, or macos releases an update). And as mentioned above, right now it is a higher priority for the project to support the latest features / platforms / versions than it is to support legacy things.

In that context, I'm tempted to do the "slightly dangerous" thing here and use "latest". The upside is that I never need to think about updates and they happen "automatically". The downside is that we are no longer in control of when we take updates, which could sometimes break CI for pull requests. However given that this is "userspace", I'm guessing breakage will be relatively uncommon.

The risk is low here, so I'm willing to roll the dice and see if "latest" is workable in this context. If it creates churn, we can re-visit.